### PR TITLE
Playing audio onscan events. Fixes #75

### LINKED ISSUE #75: Incorporate sound play during Credential Verification

### PR DESCRIPTION
**Pull request by jmcanterafonseca-iota**

### CODE REVIEW

**Comment by jmcanterafonseca-iota** [comment]
#75

File: identity-enabler/verifier-mobile-app/src/App.svelte [hunk 1]
```diff
@@ -8,7 +8,7 @@
     import FullScreenLoader from './components/FullScreenLoader.svelte';
     import { loadingScreen } from './lib/store';
     import { handleScannerData } from './lib/scan';
-    import { showAlert } from './lib/ui/helpers';
+    import { playAudio, showAlert } from './lib/ui/helpers';
```
Comment (7opf) [inline identity-enabler/verifier-mobile-app/src/App.svelte:11]: Unused import
```suggestion
    import { showAlert } from './lib/ui/helpers';
```

File: identity-enabler/verifier-mobile-app/src/pages/InvalidCredential.svelte [hunk 2]
```diff
@@ -1,7 +1,15 @@
 <script>
+import { onMount } from 'svelte';
+
     import { navigate } from 'svelte-routing';
     import { fly } from 'svelte/transition';
     import Button from '../components/Button.svelte';
+import { playAudio } from '../lib/ui/helpers';
+    const PLAY_DELAY = 500;
+    onMount(() => {
+        // We wait a little bit in order not to overlap the different aural feedback
```
Comment (7opf) [inline identity-enabler/verifier-mobile-app/src/pages/InvalidCredential.svelte:10]: typo :)
```suggestion
        // We wait a little bit in order not to overlap the different sounds
```

File: identity-enabler/deviceId-mobile-app/src/lib/ui/helpers.ts [hunk 3]
```diff
@@ -4,3 +4,14 @@ export async function showAlert(title: string, message: string) {
     const { Modals } = Plugins;
     await Modals.alert({ title, message });
 }
+
+export async function playAudio(sound: string) {
```
Comment (7opf) [inline identity-enabler/deviceId-mobile-app/src/lib/ui/helpers.ts:8]: I think we should extend this function to support a millisecond delay (default=0) as a second parameter so we don't repeat this everywhere

File: identity-enabler/holder-mobile-app/src/lib/ui/helpers.ts [hunk 4]
```diff
@@ -4,3 +4,14 @@ export async function showAlert(title: string, message: string) {
     const { Modals } = Plugins;
     await Modals.alert({ title, message });
 }
+
+export async function playAudio(sound: string) {
```
Comment (7opf) [inline identity-enabler/holder-mobile-app/src/lib/ui/helpers.ts:8]: Same comment regarding delay

File: identity-enabler/verifier-mobile-app/src/lib/ui/helpers.ts [hunk 5]
```diff
@@ -4,3 +4,14 @@ export async function showAlert(title: string, message: string) {
     const { Modals } = Plugins;
     await Modals.alert({ title, message });
 }
+
+export async function playAudio(sound: string) {
```
Comment (7opf) [inline identity-enabler/verifier-mobile-app/src/lib/ui/helpers.ts:8]: Same comment regarding delay

File: identity-enabler/deviceId-mobile-app/src/pages/Scan.svelte [hunk 6]
```diff
@@ -22,25 +23,35 @@
...
         try {
+            await playAudio('scanned');
+
             loading = true;
             let parsedData = parse(event.detail);
             VP = parsedData;
 
-            if (!VP) return showAlert();
+            if (!VP) {
+                await playAudio('invalid');
```
Comment (7opf) [inline identity-enabler/deviceId-mobile-app/src/pages/Scan.svelte:33]: Missing delay?

File: identity-enabler/deviceId-mobile-app/src/pages/Scan.svelte [hunk 7]
```diff
@@ -22,25 +23,35 @@
...
             const verificationResult = await identityService.verifyVerifiablePresentation(VP);
     
             if (verificationResult) {
+                await playAudio('valid');
+
                 showToast();
                 loading = false;
                 navigate('credential', { state: { credential: VP, save: true }});
             } else {
+                await playAudio('invalid');
```
Comment (7opf) [inline identity-enabler/deviceId-mobile-app/src/pages/Scan.svelte:47]: Missing delay?

File: identity-enabler/deviceId-mobile-app/src/pages/Scan.svelte [hunk 8]
```diff
@@ -22,25 +23,35 @@
...
+            if (!VP) {
+                await playAudio('invalid');
+                return showAlert();
+            }
 
             const identityService = ServiceFactory.get<IdentityService>('identity');
             const verificationResult = await identityService.verifyVerifiablePresentation(VP);
     
             if (verificationResult) {
+                await playAudio('valid');
```
Comment (7opf) [inline identity-enabler/deviceId-mobile-app/src/pages/Scan.svelte:41]: Missing delay?

File: identity-enabler/deviceId-mobile-app/src/pages/Scan.svelte [hunk 9]
```diff
@@ -22,25 +23,35 @@
...
                 navigate('credential', { state: { credential: VP, save: true }});
             } else {
+                await playAudio('invalid');
+
                 loading = false;
                 showAlert();
                 error.set('Invalid Data Matrix');
             }
         } catch (err) {
+            await playAudio('invalid');
```
Comment (7opf) [inline identity-enabler/deviceId-mobile-app/src/pages/Scan.svelte:54]: Missing delay?

File: identity-enabler/verifier-mobile-app/src/lib/scan.ts [hunk 10]
```diff
@@ -41,6 +42,7 @@ export async function handleScannerData(decodedText: string): Promise<void> {
             text: 'Credential verified!',
             position: 'center'
         });
+        playAudio('valid');
```
Comment (7opf) [inline identity-enabler/verifier-mobile-app/src/lib/scan.ts:45]: Missing delay?

**Review by 7opf — CHANGES_REQUESTED** [review]
Works fine, but some changes needed and other suggested.

File: identity-enabler/verifier-mobile-app/src/lib/scan.ts (hunk 10 above)
Comment (jmcanterafonseca-iota) [reply identity-enabler/verifier-mobile-app/src/lib/scan.ts:45]: not necessary as there is enough delay when processing a legit credential ...

File: identity-enabler/verifier-mobile-app/src/App.svelte (hunk 1 above)
Comment (jmcanterafonseca-iota) [reply identity-enabler/verifier-mobile-app/src/App.svelte:11]: good catch

**Pull request merged**